delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/resources/ResourceRejection.scala

```diff
@@ -0,0 +1,159 @@
+
+  /**
+    * Rejection returned when attempting to update/deprecate a resource with a different schema than the resource schema.
+    *
+    * @param id       the resource identifier
+    * @param provided the resource provided schema
+    * @param expected the resource schema
+    */
+  final case class ResourceSchemaUnexpected(id: Iri, provided: ResourceRef, expected: ResourceRef)
+      extends ResourceRejection(
+        s"Resource '$id' is not constrained by the provided schema '$provided', but by the schema '$expected'."
+      )
+
+  /**
+    * Rejection returned when attempting to create a SHACL engine.
+    *
+    * @param id      the resource identifier
+    * @param schema  the resource provided schema
+    * @param details the SHACL engine errors
+    */
+  final case class ResourceShaclEngineRejection(id: Iri, schema: ResourceRef, details: String)
+      extends ResourceRejection(s"Resource '$id' failed to produce a SHACL engine for schema '$schema'.")
+
+  /**
+    * Rejection returned when attempting to update/deprecate a resource that is already deprecated.
+    *
+    * @param id      the resource identifier
+    */
+  final case class ResourceIsDeprecated(id: Iri) extends ResourceRejection(s"Resource '$id' is deprecated.")
+
+  /**
+    * Rejection returned when a subject intends to perform an operation on the current resource, but either provided an
+    * incorrect revision or a concurrent update won over this attempt.
+    *
+    * @param provided the provided revision
+    * @param expected the expected revision
+    */
+  final case class IncorrectRev(provided: Long, expected: Long)
+      extends ResourceRejection(
+        s"Incorrect revision '$provided' provided, expected '$expected', the resource may have been updated since last seen."
+      )
+
+  /**
+    * Signals and attempt to interact with a resource that belongs to a deprecated project.
+    */
+  final case class ProjectIsDeprecated(projectRef: ProjectRef)
+      extends ResourceRejection(s"Project with label '$projectRef' is deprecated.")
+
+  /**
+    * Signals that an operation on a resource cannot be perform due to the fact that its project does not exist.
+    */
+  final case class ProjectNotFound(projectRef: ProjectRef)
+      extends ResourceRejection(s"Project '$projectRef' not found.")
+
+  /**
+    * Rejection returned when attempting to validate a resource against a schema that is deprecated.
+    *
+    * @param ref     the schema reference
+    */
+  final case class SchemaIsDeprecated(ref: ResourceRef) extends ResourceRejection(s"Schema '$ref' is deprecated.")
+
+  /**
+    * Rejection returned when attempting to validate a resource against a schema that is deprecated.
+    *
+    * @param ref     the schema reference
+    */
+  final case class SchemaNotFound(ref: ResourceRef) extends ResourceRejection(s"Schema '$ref' not found.")
```

Comment:
Not for this PR, but would be nice to consider providing the full resolution tree in this situation when resolvers would be implemented.